puppet.go

```diff
@@ -22,68 +22,119 @@ import (
 	"io"
 	"log"
 	"os/exec"
+	"strconv"
 	"strings"
+	"time"
 )
 
-func PuppetConfig(puppetParameter, puppetConf string) *GraphConfig {
-	var puppetConfArg string
-	if puppetConf != "" {
-		puppetConfArg = "--config=" + puppetConf
-	}
-
-	var cmd *exec.Cmd
-	if puppetParameter == "agent" {
-		cmd = exec.Command("puppet", "mgmtgraph", "print", puppetConfArg)
-	} else if strings.HasSuffix(puppetParameter, ".pp") {
-		cmd = exec.Command("puppet", "mgmtgraph", "print", puppetConfArg, "--manifest", puppetParameter)
-	} else {
-		cmd = exec.Command("puppet", "mgmtgraph", "print", puppetConfArg, "--code", puppetParameter)
-	}
+const (
+	PUPPET_YAML_BUFFER_SIZE = 65535
+)
 
+func runPuppetCommand(cmd *exec.Cmd) ([]byte, error) {
 	if DEBUG {
-		log.Printf("running command: %v", cmd)
+		log.Printf("Puppet: running command: %v", cmd)
 	}
 
 	stdout, err := cmd.StdoutPipe()
 	if err != nil {
-		log.Printf("Error opening pipe to puppet command: %v", err)
-		return nil
+		log.Printf("Puppet: Error opening pipe to puppet command: %v", err)
+		return nil, err
 	}
 	stderr, err := cmd.StderrPipe()
 	if err != nil {
-		log.Printf("Error opening error pipe to puppet command: %v", err)
-		return nil
+		log.Printf("Puppet: Error opening error pipe to puppet command: %v", err)
+		return nil, err
 	}
 
-	log.Println("launching puppet translator")
 	if err := cmd.Start(); err != nil {
-		log.Printf("Error starting puppet command: %v", err)
-		return nil
+		log.Printf("Puppet: Error starting puppet command: %v", err)
+		return nil, err
 	}
 
-	data := make([]byte, 65535)
-	var config GraphConfig
+	// XXX: the current implementation is likely prone to fail
+	// as soon as the YAML data overflows the buffer.
```

Comment:
On Thu, May 26, 2016 at 8:21 AM, Felix Frank <notifications@github.com>
wrote:

> We will want some loop construct that keeps reading and catenating data
> until we have it all.



Can you add this?